omniqueue/src/backends/memory_queue.rs

```diff
@@ -131,7 +131,11 @@ impl QueueConsumer for MemoryQueueConsumer {
     type Payload = Vec<u8>;
 
     async fn receive(&mut self) -> Result<Delivery, QueueError> {
-        let payload = self.rx.recv().await.map_err(QueueError::generic)?;
+        let payload = self
+            .rx
+            .recv()
+            .await
+            .ok_or_else(|| QueueError::Generic("recv failed".into()))?;
```

Comment:
I used `QueueError::Generic(mpsc::TryRecvError::Disconnected)` here instead in my local branch, not sure if it's better or worse than what you wrote though ^^

Reply:
Feel free to fix that up as you see fit in your continuing dev ;)